linkerd/app/src/identity.rs

```diff
@@ -55,7 +69,8 @@ impl Config {
                     )))
                     .push_timeout(control.connect.timeout)
                     .push(control::client::layer())
-                    .push(control::resolve::layer(dns))
+                    .push(discover)
```

Comment:
Can we abstract all of that for identity, dst and the oc collector. If so, what would a method that produces this stack return? `impl MakeService` ? I had some trouble with the typo contraints